### PR TITLE
Exclude arm64 from valid iOS simulators

### PR DESCRIPTION
**Pull request by jmagman**
Another attempt at https://github.com/flutter/flutter/pull/73458 on top of [minimum Xcode 12 version bump](https://github.com/flutter/flutter/pull/77025).

### Description
Exclude unsupported new arm64 and old i386 iOS simulators at the build setting layer.  This will allow the excluded architectures to be excluded when the project is run from Xcode, instead of just from the `flutter` command line.

Do this for plugin targets (in the podhelper) as well as in the Runner app target (Generated.xcconfig).

Update the add-to-app host apps to exclude arm64 simulators since Flutter still doesn't contain that slice.
Sample code update: https://github.com/flutter/samples/pull/646
Website docs update: https://github.com/flutter/website/pull/5174

https://github.com/flutter/flutter/issues/64502#issuecomment-755839408

Added a ios_content_validation_test case to target the simulator with `ONLY_ACTIVE_ARCH=NO` which will target all valid architectures that have not been explicitly excluded.  This test case would fail if arm64 or i386 were not excluded.

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart [hunk 1]
```diff
@@ -14,200 +14,258 @@ import '../src/darwin_common.dart';
...
-        processManager.runSync(<String>[
-          flutterBin,
-          ...getLocalEngineArguments(),
-          'create',
-          '--platforms=ios',
-          '-i',
-          'objc',
-          'hello',
-        ], workingDirectory: tempDir.path);
+  group('iOS app validation', () {
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart:17]: Most of this file is whitespace changes.

File: packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart [hunk 2]
```diff
@@ -14,200 +14,258 @@ import '../src/darwin_common.dart';
...
-          outputAppFrameworkBinary.path,
-        ]);
-        expect(grepResult.stdout, isNot(contains('matches')));
-      });
-    },
-      skip: !platform.isMacOS,
-      timeout: const Timeout(Duration(minutes: 5)),
-    );
-  }
+    testWithoutContext('build for simulator with all available architectures', () {
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart:234]: This test is new.

**Comment by jmagman** [comment]
Third time's the charm (maybe)!

**Comment by jonahwilliams** [comment]
Can you remind me what the failure was last time? Xcode version issues?

**Comment by jmagman** [comment]
> Can you remind me what the failure was last time? Xcode version issues?

https://github.com/flutter/flutter/pull/73807#issuecomment-759159802

Xcode 11 had an allow list `VALID_ARCHS` build setting which I had used before ARM was a thing to indicate that plugins shouldn't build for the 32-bit i386 simulator.
https://github.com/flutter/plugins/blob/bc2514ebd824adb1c85709e53bc57235af7b5212/packages/device_info/device_info/ios/device_info.podspec#L21
```ruby
s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES', 'VALID_ARCHS[sdk=iphonesimulator*]' => 'x86_64' }
```

Xcode 12 has a new `EXCLUDED_ARCHS` build setting which is the deny list version of `VALID_ARCHS`, which the plugin template now uses instead #64504.  But there are plenty of plugins that use the older style `VALID_ARCHS` plugin template.

When both were present, Xcode 11 had a little freak out, which was the cause for the last two reverts.
```
stdout:     warning: There are no architectures to compile for because all architectures in VALID_ARCHS (x86_64) are also in EXCLUDED_ARCHS (x86_64, i386). (in target 'device_info' from project 'Pods')
```

That error doesn't even make sense--the `EXCLUDED_ARCHS` build setting in this PR doesn't contain `x86_64`.

But now the devicelab is on Xcode 12 and it's the minimum version the tool allows for developers.  Xcode 12 handles these build settings, so with this PR even when plugins have `VALID_ARCHS` set, it prefers `EXCLUDED_ARCHS` and excludes `arm64` and `i386`.  Once the Flutter.framework ships with an arm64 simulator, we can remove `arm64` from that list.

**Review by jonahwilliams — APPROVED** [review]
LGTM!

**Pull request merged**